Terminal.Gui/View/View.Drawing.cs

```diff
@@ -1,280 +1,282 @@
+    ///         The view will only be drawn if it is visible, and has any of <see cref="NeedsDraw"/>,
+    ///         <see cref="SubViewNeedsDraw"/>,
+    ///         or <see cref="NeedsLayout"/> set.
+    ///     </para>
+    ///     <para>
+    ///         See the View Drawing Deep Dive for more information: <see href="https://gui-cs.github.io/Terminal.GuiV2Docs/docs/drawing.html"/>.
     ///     </para>
     /// </remarks>
-    public void Clear ()
+    public void Draw ()
     {
-        if (Driver is null)
+        if (!CanBeVisible (this))
         {
             return;
         }
 
-        // Get screen-relative coords
-        Rectangle toClear = ViewportToScreen (Viewport with { Location = new (0, 0) });
-
-        Rectangle prevClip = Driver.Clip;
+        Region? saved = GetClip ();
 
-        if (ViewportSettings.HasFlag (ViewportSettings.ClearContentOnly))
+        // TODO: This can be further optimized by checking NeedsDraw below and only clearing, drawing text, drawing content, etc. if it is true.
+        if (NeedsDraw || SubViewNeedsDraw)
         {
-            Rectangle visibleContent = ViewportToScreen (new Rectangle (new (-Viewport.X, -Viewport.Y), GetContentSize ()));
-            toClear = Rectangle.Intersect (toClear, visibleContent);
-        }
+            // Draw the Border and Padding.
+            // We clip to the frame to prevent drawing outside the frame.
+            saved = ClipFrame ();
+            DoDrawBorderAndPadding ();
+            SetClip (saved);
 
-        Attribute prev = Driver.SetAttribute (GetNormalColor ());
-        Driver.FillRect (toClear);
-        Driver.SetAttribute (prev);
+            // Draw the content within the Viewport
+            // By default, we clip to the viewport preventing drawing outside the viewport
+            // We also clip to the content, but if a developer wants to draw outside the viewport, they can do
+            // so via settings. SetClip honors the ViewportSettings.DisableVisibleContentClipping flag.
+            // Get our Viewport in screen coordinates
 
-        Driver.Clip = prevClip;
-    }
+            saved = ClipViewport ();
 
-    /// <summary>Fills the specified <see cref="Viewport"/>-relative rectangle with the specified color.</summary>
-    /// <param name="rect">The Viewport-relative rectangle to clear.</param>
-    /// <param name="color">The color to use to fill the rectangle. If not provided, the Normal background color will be used.</param>
-    public void FillRect (Rectangle rect, Color? color = null)
-    {
-        if (Driver is null)
-        {
-            return;
+            // Clear the viewport
+            // TODO: Simplify/optimize SetAttribute system.
+            DoSetAttribute ();
+            DoClearViewport ();
+
+            // Draw the subviews
+            if (SubViewNeedsDraw)
+            {
+                DoSetAttribute ();
+                DoDrawSubviews ();
```

Comment:
Subviews are drawn before the superview itself? Is this correct? `DoDrawContent` should be at first and then the subviews. Am I wrong?

Reply:
Please read `drawing.md`.

With non-rectangular Clip, views "above" now get drawn first. 

Then views "below" get drawn, but don't overwrite because clipping is actually doing its job.

Reply:
Got it and thus only the visible sides are drawn. That's really impressive and have to be well articulated. When UpdateScreen is call all contents must be already defined and no background call mustn't corrupt it, with exception of sixel, of course. Good work. Thanks.